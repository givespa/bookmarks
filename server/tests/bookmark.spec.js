const request = require('supertest');
const app = require('../app');

describe('Bookmark API', () => {
    it('should return the bookmarks', async () => {
        const res = await request(app).get('/bookmarks');
        expect(res.statusCode).toBe(200);
        expect(res.body).toMatchSnapshot();
    });

    it('should add a new bookmark', async () => {
        const res = await request(app)
            .post('/bookmarks')
            .set('Content-Type', 'application/json')
            .send(
                '{ "title": "Google", "link": "https://www.google.com" }',
            );;
        expect(res.statusCode).toBe(200);
        expect(res.body).toMatchSnapshot();
    });
});
