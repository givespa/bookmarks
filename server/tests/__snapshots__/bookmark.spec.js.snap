// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark API should add a new bookmark 1`] = `Object {}`;

exports[`Bookmark API should return the bookmarks 1`] = `Array []`;
